packages/amplify-codegen/src/commands/models.js

```diff
@@ -99,19 +101,48 @@ function loadSchema(apiResourcePath) {
   throw new Error('Could not load the schema');
 }
 
-function getModelOutputPath(context) {
+async function getModelOutputPath(context) {
   const projectConfig = context.amplify.getProjectConfig();
+  if(!projectConfig.frontend){
+    throw new Error('Frontend type is not configured.');
+  }
+
+  if(projectConfig.modelgen && projectConfig.modelgen.outputPath){
+    context.print.success('\nThe output path for modelgen is already setup. You can edit the path in .config/project-config.json\n');
+    return projectConfig.modelgen.outputPath;
+  }
+
+  let defaultPath = '.';
   switch (projectConfig.frontend) {
     case 'javascript':
-      return 'src';
+      defaultPath = 'src/models';
+      break;
     case 'android':
-      return projectConfig.android && projectConfig.android.config && projectConfig.android.config.ResDir
+      defaultPath = projectConfig.android && projectConfig.android.config && projectConfig.android.config.ResDir
         ? path.normalize(path.join(projectConfig.android.config.ResDir, '..', 'java'))
         : path.join('app', 'src', 'main', 'java');
+      break;
     case 'ios':
-      return 'amplify/generated/models';
-    default:
-      return '.';
+      defaultPath =  'amplify/generated/models';
+      break;
   }
+
+  let outputPath = defaultPath;
+  if(!context.parameters.options.yes){
```

Comment:
Keep in mind that the current version of the Gradle plugin _won't_ be provided `--yes`. So, there are four scenarios you need to account for:
1. Old CLI version, old plugin (current behavior, it kinda works.)
2. Old CLI version, new plugin (the user doesn't update CLI, but does have your plugin updates.)
3. New CLI version, old plugin (user updates the CLI, but uses the current plugin version.)
4. New CLI version, new plugin (this one is easy, this is what you _want_ to happen. I assume this will work.)

What happens in cases (2), (3)? If it breaks, how will you convey a message to the user, to tell them to update?

Reply:
Good points to consider. We have no problems on (1) and (4). I just figure out how to test it locally and try scenario (2) and (3). I get everything right in (2). (adding extra prameters will not impact the old modelgen).  However, I get it stuck in (3) since the command line executed by old plugin does not include `--yes` and will stop on the inquiry and no models are generated. One way to fix this problem is that we add a warning print in the new modelgen code to remind users of getting plugin updated.

Reply:
I just have a good idea about this issue. Just switch the two operations and there is no need to change code in SDK libraries. More specifically, the question will only pop up when `--configpath` flag is passed(I change the flag name since the function of flag is changed). And when no flags exist in cmd, it simply generate models with default path just the same as the old one and no questions will pop up, while the only difference is that the output path is stored in `amplify/.config/project-meta.json`.  Since no code changes to both android and ios plugin and both old and new version of CLI generate the same output on running `amplify codegen models`, all the issues are perfectly resolved

Reply:
I run the API test in android studio with amplify tools plugin and find this method much better than the previous one, which is really beyond my expectation. You can have customized output path this time even by using gradle plugins. Simply changing the `modelgen.outputPath` field in `project-meta.json` and users are all set.